ngraph/test/onnx/onnx_editor.cpp

```diff
@@ -0,0 +1,145 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <algorithm>
+#include "gtest/gtest.h"
+
+#include "ngraph/file_util.hpp"
+#include "ngraph/op/util/op_types.hpp"
+#include "onnx_import/default_opset.hpp"
+#include "onnx_import/editor/editor.hpp"
+#include "onnx_import/onnx.hpp"
+#include "util/test_control.hpp"
+
+NGRAPH_SUPPRESS_DEPRECATED_START
+
+using namespace ngraph;
+
+static std::string s_manifest = "${MANIFEST}";
+
+namespace
+{
+    using InputTypePred = std::function<bool(const std::shared_ptr<ngraph::Node>)>;
+
+    // A higher order factory function that produces predicates bound to a particular element type
+    InputTypePred element_type_is(const element::Type et)
+    {
+        return [et](const std::shared_ptr<ngraph::Node> input) {
+            return std::dynamic_pointer_cast<onnx_import::default_opset::Parameter>(input)
+                       ->get_element_type() == et;
+        };
+    }
+} // namespace
+
+NGRAPH_TEST(onnx_editor, single_input_type_substitution)
+{
+    // the original model contains 2 inputs with i64 data type and one f32 input
+    onnx_import::ONNXModelEditor editor{
+        file_util::path_join(SERIALIZED_ZOO, "onnx/model_editor/add_abc.prototxt")};
+
+    editor.set_input_types({{"A", element::i64}});
+
+    const auto function = onnx_import::import_onnx_model(editor);
+
+    const auto graph_inputs = function->get_parameters();
+
+    const auto float_inputs_count = std::count_if(
+        std::begin(graph_inputs), std::end(graph_inputs), element_type_is(element::f32));
+
+    const auto integer_inputs_count = std::count_if(
+        std::begin(graph_inputs), std::end(graph_inputs), element_type_is(element::i64));
+
+    EXPECT_EQ(float_inputs_count, 0);
+    EXPECT_EQ(integer_inputs_count, 3);
```

Comment:
I think that it is worth to add one test which confirm that exactly type of "A" parameter was changed. I can be sth like (not tested):
```
    const auto param_a =
        std::find_if(graph_inputs.begin(), graph_inputs.end(), [](const std::shared_ptr<Parameter>& param) {
            return std::string{param->get_friendly_name()} == "A";
        });
EXPECT_EQ(param_a->get_element_type(), element::i64);
```